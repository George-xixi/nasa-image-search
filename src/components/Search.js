import React from "react";
import "../styles/search.css";

function Search() {
  return (
    <>
      <input className="search-input" type="text" data-testid="search-input" />
    </>
  );
}

export default Search;
