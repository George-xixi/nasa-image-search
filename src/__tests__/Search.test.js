import { render, screen } from "@testing-library/react";
import Search from "../components/Search";

describe("Search", () => {
  it("renders Search component", () => {
    const { asFragment } = render(<Search />);
    const inputElement = screen.getByTestId("search-input");

    expect(asFragment()).toMatchSnapshot();
    expect(inputElement).toHaveAttribute("class", "search-input");
  });
});
