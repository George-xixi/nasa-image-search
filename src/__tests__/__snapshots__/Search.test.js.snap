// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders Search component 1`] = `
<DocumentFragment>
  <input
    class="search-input"
    data-testid="search-input"
    type="text"
  />
</DocumentFragment>
`;
